### DISCUSSION TITLE
Has anyone successfully hosted Etherpad Lite on Azure?

### DISCUSSION THREAD

**Discussion by alexeisenhart** [ether/etherpad-lite#4798] [Q&A]
I've been struggling to get an instance of Etherpad-lite deployed to Azure for the last 2 days. I've tried both Linux and Windows hosting.

Linux: I can run the site in Ubuntu in a Hyper-V VM, but deploying these files to Azure has lead me into issues all day.

Windows: I cannot run locally due to issue #4787, but I've tried to work around it to no avail.

Docker: I CAN deploy the docker image, but then I can't modify the settings files. I'm aware of the documentation here (https://github.com/ether/etherpad-lite/blob/develop/doc/docker.md) for building a custom docker container, but I'm concerned that this isn't going to work because of all of the issues that I've had with both the Linux and Windows environments.

Has anyone successfully deployed an instance to Azure? The Linux issue is failing because it can't find npm-cli.js, but I think that's an issue with symbolic links in Azure, but I'm not sure.

My background is in ASP.NET development and Windows hosting, so Node.js and Azure are both new to me. If I started from scratch tomorrow, would anyone be able to help me debug the issues that I'm running into? I'm open to any advice or ideas. I am not limited by the hosting platform: Windows, Linux or Docker would be fine with me.

**Comment by JohnMcLear** [comment]
Fix landing in an hour.  Apologies.

**Reply by alexeisenhart** [reply]
No problem! And thank you so much for your fast response and turn around! By all means, I'm grateful to the developers and maintainers of this software package. 

I'm having all sorts of other unrelated issues with the Linux installation path as well, but I think they're Azure related and it's frustrating. Do you know if anyone has installed Etherpad-Lite on Azure?

I'll give the Windows path a fresh try after you make the check in.

**Reply by JohnMcLear** [reply]
I just pushed the fix to develop branch so it's available there.

I also added CI build for Windows so this wont happen again.

I am pretty sure Etherpad on Azure is fine :)
